core/src/toga/documents.py

```diff
@@ -79,6 +80,11 @@ def title(self) -> str:
         """
         return f"{self.document_type}: {self.path.stem if self.path else 'Untitled'}"
 
+    @property
+    def is_modified(self):
```

Comment:
My choice around having `touch()`, and no writable property was deliberate - once a document is "dirty", the only way to "undirty" it is to save the document, so having the property writable doesn't make sense. Plus, it allows for `touch()` to be defined in a way that it can be an event handler (rather than requiring every user to define a handler for this common use case - see the documentapp example for usage). 

I think I agree about the `is_` prefix, though; I'll remove that.

Reply:
Thinking about this some more (and checking with some apps), "undo all changes" does revert the "dirty" state, so making `modified` a writable property makes sense. However, I think it's worth keeping the `touch()` method, specifically so there's a version that can be used as an event handler (since it's such an obvious use case).